test/unit/shmem_team_ptr.c

```diff
@@ -0,0 +1,162 @@
+/*
+ *  Copyright (c) 2022 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <shmem.h>
+
+int main(void) {
+    int i, n, errors = 0;
+    int me, team_me, npes;
+
+    static int shr_world_data, shr_team_data = -1;
+    int * shr_heap;
+
+    shmem_init();
+
+    me = shmem_my_pe();
+    npes = shmem_n_pes();
+
+    shr_world_data = me;
+
+    shmem_barrier_all();
+
+    /* Check shmem_team_ptr on data segment */
+    for (i = n = 0; i < npes; i++) {
+        int * world_ptr = (int *) shmem_team_ptr(SHMEM_TEAM_WORLD, &shr_world_data, i);
+
+        if (world_ptr != NULL) {
```

Comment:
IIUC, this if condition is not mutually exclusive to the `else if` condition below. It is just a thought that this structure could have been written like this:
```
if (i == me && world_ptr != NULL)
{
 ...
} else {
 Error..
}
```